### PR TITLE
Bump netlayer to use tor binary from tor browser v10.0

### LINKED ISSUE #4593: Extend build script with ability to verify tor binaries delivered with netlayer
Add ability to check that the tor binaries delivered by `netlayer` match the officially distributed ones.

For official tor singatures and hashes, see [v9.5.4](https://dist.torproject.org/torbrowser/9.5.4/) and [v10.0](https://dist.torproject.org/torbrowser/10.0/).

### PR DESCRIPTION
**Pull request by cd2357**
Variant of #4601 which is based on `tor-browser` v10.0 (instead of v9.5.4 used in that PR).

Use a `netlayer` version that includes tor binaries extracted from the latest tor browser [v10.0](https://dist.torproject.org/torbrowser/10.0/).

For simplicity:
- use [netlayer version cc80787](https://jitpack.io/#cd2357/netlayer/cc80787) (based on commit `cc80787` from [this branch](https://github.com/cd2357/netlayer/commits/upgrade-tor-binary-0.4.3.9-on-v0.6.8))
  - the referenced branch = previously used `netlayer` v0.6.8 + the following change to `tor-binary`
- above `netlayer` bumps `tor-binary` dependency to [3dbd395](https://jitpack.io/#cd2357/tor-binary/3dbd395) (based on commit `3dbd395` from [this branch](https://github.com/cd2357/tor-binary/commits/upgrade-tor-10.0))
  - the referenced branch = previously used `tor-binary` dependency + change A + change B
    - change A: extract tor binaries from `tor-browser` v10.0 (instead of 9.5.3 used previously)
    - change B: update the extraction and build process to check if the `SHA-256` hashes of the `tor-browser` binaries match [the official ones](https://dist.torproject.org/torbrowser/10.0/sha256sums-signed-build.txt) (instead of `SHA-512` hashes used previously, which are not published in the official tor repo anymore)
      - this ensures the build only succeeds if the downloaded `tor-browser` binaries (used to extract the tor binaries) match the official checksums

`tor-browser` v10.0 updates the tor binaries to v0.4.4.5 as per [the tor browser v10 changelog](https://gitweb.torproject.org/builders/tor-browser-build.git/plain/projects/tor-browser/Bundle-Data/Docs/ChangeLog.txt?h=maint-10.0). Currently, Bisq uses the tor binary extracted from `tor-browser` v9.5.3, which is tor v0.4.3.6 (as per [the tor browser v9.5 changelog](https://gitweb.torproject.org/builders/tor-browser-build.git/plain/projects/tor-browser/Bundle-Data/Docs/ChangeLog.txt?h=maint-9.5))

In other words:
- compared to PR #4601 
  - this PR upgrades tor from v0.4.3.6 to v0.4.4.5 (see [here](https://gitweb.torproject.org/tor.git/plain/ReleaseNotes) for tor changes)
- compared to the latest master
  - the above note, plus
  - ensures that the used tor binaries were extracted from verified `tor-browser` packages, using their `SHA-256` hashes

Fixes #4593

### CODE REVIEW

**Review by chimp1984 — APPROVED** [review]
utACK

**Comment by chimp1984** [comment]
I dont have any  preference here related to the other PR. @cd2357 Which do you recommend?

**Comment by cd2357** [comment]
The other PR is more conservative (tor browser 9.5.3 -> 9.5.4), this one more aggressive (9.5.3 -> 10.0).

But since tor browser 9.5.4 will be the last one from the 9.5 branch (as per [release notes](https://blog.torproject.org/new-release-tor-browser-954)), it means only 10.x will receive patches and updates from now on.

Which means we'll have to switch to the 10.0 binaries sooner or later anyway. So we might as well do it now.

So I'd suggest this PR over #4601.

**Comment by ripcurlx** [comment]
@cd2357 On which OS did you test the new binaries?

Works on macOS 10.15.6

**Comment by cd2357** [comment]
> @cd2357 On which OS did you test the new binaries?

I tested this PR on macOS 10.15.6 + Windows 10 + Debian 10.6, where tested = started the desktop app, synchronized DAO to latest state, browsed around a bit.

**Comment by freimair** [comment]
the tor browser version is of no interest at all to bisq or the netlayer or the tor-binary project. The only reason to use the tor browser binaries in the first place was that the tor guys released signed versions of these binaries - hence, we have a trusted source of tor binaries. There has not been any other source of plain tor binaries. This is also why we do not have tor for ARM in there yet, despite https://github.com/JesusMcCloud/tor-binary/pull/3.

That being said, @cd2357 can we trust you that you verified every and all signatures prior to injecting the sha256 sums? did anyone check if cd2357 did?

**Comment by cd2357** [comment]
> the tor browser version is of no interest at all to bisq or the netlayer or the tor-binary project. The only reason to use the tor browser binaries in the first place was that the tor guys released signed versions of these binaries - hence, we have a trusted source of tor binaries.

Yes.

Currently there was no way to check if the tor binaries used in Bisq were the ones extracted from authenticated tor browser binaries, because:
- current tor binaries used in Bisq are extracted from tor-browser 9.5.3 (for which binaries or hashes are not provided anymore on the official website)
- the build process which processed those tor-browsers to extract the tor binaries relied on `SHA-512`, and there are no published `SHA-512` hashes for any official tor-browser binary

Therefore, to establish some sort of proof that our jars are built from binaries from the official tor-browser binaries, I forked `tor-binary` and `netlayer` (links to branches and jitback builds in PR description above), plus modified the process to check the published `SHA-256` hashes.

> That being said, @cd2357 can we trust you that you verified every and all signatures prior to injecting the sha256 sums? did anyone check if cd2357 did?

Don't trust, verify :)

- Published signed hashes: https://dist.torproject.org/torbrowser/10.0/sha256sums-signed-build.txt
- Signature of published signed hashes: https://dist.torproject.org/torbrowser/10.0/sha256sums-signed-build.txt.asc
- Public key for checking the signature: https://support.torproject.org/tbb/how-to-verify-signature/
- Committed `SHA-256` hashes: https://github.com/cd2357/tor-binary/commit/3dbd395e8c5557dee14103ba57f88c8fea155a59

The commited hashes should match the ones from the published hash list, which should match the list signature linked above.

**Review by ripcurlx — APPROVED** [review]
ACK

Tested it on macOS Mainnet and verified all hashes with the public hashes provided by the tor project
